civictechprojects/views.py

```diff
@@ -977,6 +1089,116 @@ def update_project_timestamp(request, project):
+
+    group = Group.objects.get(id=group_id)
+    if not is_creator(user, group):
+        return HttpResponse(status=403)
+
+    body = json.loads(request.body)
+    project = Project.objects.get(id=body['projectId'])
+    message = body['message']
+    project_relation = ProjectRelationship.create(group, project, message)
+    project_relation.save()
+    send_group_project_invitation_email(project_relation)
+    return HttpResponse(status=200)
+
+
+# TODO: Pass csrf token in ajax call so we can check for it
+@csrf_exempt
+def accept_group_invitation(request, invite_id):
+    # Redirect to login if not logged in
+    if not request.user.is_authenticated():
+        return redirect(section_url(FrontEndSection.LogIn, {'prev': request.get_full_path()}))
+
+    project_relation = ProjectRelationship.objects.get(id=invite_id)
+    project = project_relation.relationship_project
+    about_project_url = section_url(FrontEndSection.AboutProject, {'id': str(project.id)})
+    if project_relation.is_approved:
+        messages.add_message(request, messages.ERROR, 'The project is already part of the group.')
+        return redirect(about_project_url)
+
+    user = get_request_contributor(request)
+    if is_co_owner_or_owner(user, project):
+        # Set approved flag
+        project_relation.is_approved = True
+        project_relation.save()
+        update_project_timestamp(request, project)
+        if request.method == 'GET':
+            messages.add_message(request, messages.SUCCESS, 'Your project is now part of the group ' + project_relation.relationship_group.group_name)
+            return redirect(about_project_url)
+        else:
+            return HttpResponse(status=200)
+    else:
+        messages.add_message(request, messages.ERROR, 'You do not have permission to accept this group invitation.')
+        return redirect(about_project_url)
+
+
+# TODO: Pass csrf token in ajax call so we can check for it
+@csrf_exempt
+def reject_group_invitation(request, invite_id):
+    # Redirect to login if not logged in
+    if not request.user.is_authenticated():
+        return redirect(section_url(FrontEndSection.LogIn, {'prev': request.get_full_path()}))
+
+    project_relation = ProjectRelationship.objects.get(id=invite_id)
+    project = project_relation.relationship_project
+    about_project_url = section_url(FrontEndSection.AboutProject, {'id': str(project.id)})
+    if project_relation.is_approved:
+        messages.add_message(request, messages.ERROR, 'The project is already part of the group.')
+        return redirect(about_project_url)
+
+    user = get_request_contributor(request)
+    if is_co_owner_or_owner(user, project):
+        project_relation.delete()
+        update_project_timestamp(request, project)
+        if request.method == 'GET':
+            # messages.add_message(request, messages.SUCCESS, 'Your project is now part of the group ' + project_relation.relationship_group.group_name)
```

Comment:
Is this something we might be able to make for this release?

Reply:
I'm not sure what a good message would be for the reject case, so I put a TODO in its place for now.